### ISSUE TITLE
sys.elasped or sys.elapsedTime?

### ISSUE THREAD

**Issue by apblack** [gracelang/language#20]
I noticed that Kim uses `sys.elapsed` in his book.  So if we change the name to `elapsedTime`, the book will need to be changed.

I recently removed `cputime` from my development branch — it wasn't implementable in the JS backend, and attempts to use performance.now were just breaking things.  I also provided elapsedTime as a synonym for elapsed, intending to deprecate elapsed eventually.

I think that `elapsedTime` is the preferable name, but wanted to ask.

**Comment by KimBruce** [comment]
That's a fairly easy change to make, so I won't be too upset (at least if
you do if by mid-August).

**Comment by kjx** [comment]
> On 7/08/2015, at 12:01pm, Kim Bruce <notifications@github.com> wrote:
> 
> That's a fairly easy change to make, so I won't be too upset (at least if you do if by mid-August).


If we’re talking “mid-August” then what about the other, bigger things we’re thinking of?


This is another related question - what kind of things are we planning to talk over in sept?

the big things like classes, inheritance / dialects, annotations, …


or just tidying up details or working out which details to tidy up?

James

**Comment by KimBruce** [comment]
I'm assuming (hoping?) there will be very few, relatively small changes to
the language (or at least minigrace) between now and the end of my fall
semester.  In particular, any substantial changes we make in Claremont
won't go live until after the semester is over.

**Comment by kjx** [comment]
> On 7/08/2015, at 12:14pm, Kim Bruce <notifications@github.com> wrote:
> 
> I'm assuming (hoping?) there will be very few, relatively small changes to
> the language (or at least minigrace) between now and the end of my fall
> semester. In particular, any substantial changes we make in Claremont
> won't go live until after the semester is over.

OK well thats a reasonable approach

I guess undotted (class as methods) from dotted (Classes as objects) is the big change we could consider
(give that andrew’s mg has “factory methods” we know stuff is there and should work)

J

**Comment by apblack** [comment]
`elaspsedTime` is live now.

As for the schedule for making changes: I won't be teaching with Grace in the Fall, so probably won't have time to do very much.  But I don't see that this is relevant.  Kim, you can freeze a version of _Kernan_ or _minigrace_ to teach with whenever you like, and also a version of the spec.  This should not stop us from making forward looking changes.

In my teaching, one of the frustrations that the students are expressing is that the spec is not consistent with the implementation — it is either ambiguous, or just says something different.   So one of my motivations for posting all of these issues is to update the spec.  I also think that this is important for publicity.  I don't like the idea of saying that we can't do that while Kim is teaching.

**Comment by KimBruce** [comment]
I agree that freezing should be fine (though I hope I can get help fixing
problems that show up during the course of the term).  I agree that the
spec needs immediate attention.  Many of those fixes/clarifications have
little or no impact on students.  However, if we are going to make major
changes -- or explore them to see how they work out -- we might want to
give the new language a distinct name (Tim was planning on that for his
work, for example) to keep from confusing anyone following us.

**Comment by apblack** [comment]
I don't like the idea of a new name.  But we might declare what we have now to be Grace 1.0, and start thinking about Grace 2.0.
